docs/content/new-tutorial/building-an-asset-graph.mdx

```diff
@@ -0,0 +1,221 @@
+If this syntax is too magical for you, you can explicitly define an asset's dependencies a parameter called "inputs" in the @asset decorator. The explicit dependency syntax is also helpful when referencing assets from other asset groups. TODO: Add link to this.
+</Note>
+
+
+### Context and logging
+
+When defining an asset, you can optionally create a `context` parameter for the function. The `context` argument contains information about the asset, such as the asset's name, the run ID, and the logger.
+
+Using the `context` argument gives you access to features. With the `context.log` object, you can print text that will appear with the job's run in the Dagster UI. With this, you can log the progress of your asset like below.
+
+```python
+import pandas as pd
+
+@asset
+def hackernews_topstories(context, top_100_newstories):
+    results = []
+    for item_id in top_100_newstories:
+        item = requests.get(f"https://hacker-news.firebaseio.com/v0/item/{item_id}.json").json()
+        results.append(item)
+
+        if len(results) % 20 == 0:
+            context.log.info(f"Got {len(results)} items so far.")
+
+    df = pd.DataFrame(results)
+
+    return df
+```
+
+For more information about loggers, such as the different types of logging, you can read the [logging documentation](/concepts/logging/loggers).
+
+---
+
+## Creating and exporting the visualization asset
+
+The last asset you'll add is the visualization asset. This asset will take the dataframe asset and create a visualization of the data. You can copy most of the last chunk of code from the original pipeline. Once again, we'll also add the relevant imports.
+
+```python
+import matplotlib.pyplot as plt
+import base64
+from io import BytesIO
+from wordcloud import STOPWORDS, WordCloud
+from dagster import asset
+
+@asset
+def hackernews_topstories_word_cloud(context, hackernews_topstories):
+    stopwords = set(STOPWORDS)
+    stopwords.update(["Ask", "Show", "HN"])
+    titles_text = " ".join([str(item) for item in hackernews_topstories["title"]])
+    titles_cloud = WordCloud(stopwords=stopwords, background_color="white").generate(titles_text)
+
+    # Generate the word cloud image
+    plt.figure(figsize=(8, 8), facecolor=None)
+    plt.imshow(titles_cloud, interpolation="bilinear")
+    plt.axis("off")
+    plt.tight_layout(pad=0)
+
+    # Save the image to a buffer
+    buffer = BytesIO()
+    plt.savefig(buffer, format="png")
+    image_data = base64.b64encode(buffer.getvalue())
+
+    return image_data
+```
+
+Right now, materializing the asset will save the file to local storage. In a real-world scenario, you would save the file to a cloud storage bucket. In a later section, you'll learn how to tell Dagster where to store the data by using IO managers.
+
+### Metadata and previews
+
+As mentioned, the asset is materialized in storage. You can add metadata to help users explore the asset and understand it. For example, you can add a preview of the asset to the metadata.
```

Comment:
i think this would be a good place to give a little more explanation. 

1. Can we help them make the leap as to why adding a preview is useful because the asset is materialized in storage? 
2. Can we add some quick bullet points about other ways that Metadata is useful: This information can be whatever you want, but possible cases include:

keeping track of the team responsible for maintaining a asset or pipeline
linking to documentation or other resources
displaying the git hash corresponding to the current job definition
where it is stored

Reply:
+1